velox/connectors/Connector.h

```diff
@@ -307,6 +307,14 @@ class ConnectorQueryCtx {
     return planNodeId_;
   }
 
+  void setCastStringToDateIsIso8601(bool flag) {
```

Comment:
This doesn't look right. Shouldn't it be part of the config?

Reply:
This doesn't feel right. HiveConnector should define its own configuration properties and use these. We should not assume that castStringToDateIsIso8601_ config applies to all connectors.

I'm still trying to understand why do you need this. I don't see tests for both cases when castStringToDateIsIso8601_ is true and when it is false. Am I missing something?

Should we add a configuration property to velox/connectors/hive/HiveConfig.h?

Reply:
@rui-mo At a high level, engine configs are separate from connector config. We should not leak engine-specific logic into connectors. I wonder why do we pass partition values in HiveConnectorSplit as string. Should we pass these as values of proper types (e.g. ConstantVectors)? Will this remove the need to converting from strings to proper types? 

I also see that applyPartitionFilter may need to be updated to handle DATE type.

Reply:
Another option might be for the connector to access cast logic in the engine via ExpressionEvaluator, i.e. construct and evaluate a CAST expression.

Reply:
Masha, thank you for the suggestion. For option 1, Velox now uses a map as below to store partition keys in [HiveConnectorSplit.h](https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/HiveConnectorSplit.h#L30-L31).
 > const std::unordered_map<std::string, std::optional<std::string>> partitionKeys

Seems `const std::unordered_map<std::string, core::ConstantTypedExpr>
      partitionKeys` can help remove the cast process. But I'm not sure if there's any special consideration in Velox to use string type. In Gluten, we need to convert literals to string type to create partition keys, and in Velox strings are converted to variants. If ConstantTypedExpr is used, the intermediate string could be removed.

Reply:
> const std::unordered_map<std::string, core::ConstantTypedExpr> partitionKeys

This could work. Would you like to prototype this change? CC: @Yuhta 

Also, wondering if any changes are needed to support writing to partitioned tables with DATE partition keys.